new-era-commands/slash/search.js

```diff
@@ -0,0 +1,67 @@
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  SlashCommandBuilder,
+} = require("discord.js");
+
+const searchSites = {
+  google: {
+    name: "Google",
+    url: "https://www.google.com/search?q=",
+    description: "Search for terms in Google",
+  },
+  top: {
+    name: "TOP",
+    url: "https://www.google.com/search?q=site:theodinproject.com+",
+    description: "Search for terms in The Odin Project website with Google",
+  },
+  mdn: {
+    name: "MDN",
+    url: "https://developer.mozilla.org/en-US/search?q=",
+    description: "Search for terms in MDN",
+  },
+};
+
+const createCommand = (site) => (command) =>
+  command
+    .setName(site.name.toLowerCase())
+    .setDescription(site.description)
+    .addStringOption((option) =>
+      option
+        .setName("prompt")
+        .setDescription("term to search")
+        .setRequired(true),
+    )
+    .addUserOption((option) =>
+      option.setName("user").setDescription("user to ping"),
+    );
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName("search")
+    .setDescription("Search in a specific site")
+    .addSubcommand(createCommand(searchSites.google))
+    .addSubcommand(createCommand(searchSites.mdn))
+    .addSubcommand(createCommand(searchSites.top)),
```

Comment:
```suggestion
  data: searchSlashCommand,
```
Which would then mean the exported object only needs to contain this as the `.data` value.

If people *really* want to have all the logic housed directly within the exported object, then an IIFE can be used I guess.